### PR TITLE
Transition to SendGrid for transactional emails

### PR DESCRIPTION
**Pull request by alistairclark**
To send yourself an email follow these steps.

1. Run `deploy_to_aws.py` so that the `EmailTemplates` dynamodb table is created
2. Add an item in this table with the following content
```
{
  "id": {
    "S": "test"
  },
  "bcc": {
    "L": [
      {
        "S": "thiscovery_dev@email.com"
      }
    ]
  },
  "cc": {
    "L": [
      {
        "S": "this_researcher@email.com"
      }
    ]
  },
  "contact_properties": {
    "L": []
  },
  "created": {
    "S": "2023-04-12 09:28:26.219073+01:00"
  },
  "custom_properties": {
    "L": [
      {
        "M": {
          "name": {
            "S": "first_name"
          },
          "required": {
            "BOOL": true
          }
        }
      }
    ]
  },
  "details": {
    "NULL": true
  },
  "from": {
    "S": "Thiscovery <thiscovery@thisinstitute.cam.ac.uk>"
  },
  "modified": {
    "S": "2023-04-12 09:28:26.219073+01:00"
  },
  "project_task_id": {
    "N": "123"
  },
  "project_task_name": {
    "S": "some random project name"
  },
  "sendgrid_template_id": {
    "S": "d-f837734467f8422ca3886d17ed297b23"
  },
  "type": {
    "S": "templates"
  }
}
```
3. Run `python utils/test_email.py --to_recipient_email {your_email_address}  --template_name test --custom_properties '{"first_name": "{your_first_name}"}'`

### CODE REVIEW

File: admin_tasks/add_email_template_to_dynamodb.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 #
 """
 This script parses a transactional email template preview page and adds the
-template to Dynamodb table HubspotEmailTemplates
+template to Dynamodb table EmailTemplates
```
Comment (this-andy) [inline admin_tasks/add_email_template_to_dynamodb.py:20]: Can you take a look at Confluence page - https://thiscovery.atlassian.net/l/cp/vmq3BPs1 
Section entitled 'Changes to templates DynamoDB table' and take account of first point there.  I don't think it's strictly necessary yet, but might as well have it in place now.

File: tests/unit_tests/test_transactional_email.py [hunk 2]
```diff
@@ -269,74 +215,74 @@ def test_12_send_transactional_email_api_locally(self):
         )
         self.assertEqual(HTTPStatus.NO_CONTENT, result["statusCode"])
```
Comment (this-andy) [inline tests/unit_tests/test_transactional_email.py:216]: Is this correct?  All the other tests use HTTPStatus.ACCEPTED

Comment (alistairclark) [reply tests/unit_tests/test_transactional_email.py:216]: So the SendGrid api returns ACCEPTED, but this is testing the `send_transactional_email` which returns NO_CONTENT.

File: admin_tasks/add_email_template_to_dynamodb.py (hunk 1 above)
Comment (alistairclark) [reply admin_tasks/add_email_template_to_dynamodb.py:20]: Makes sense, but not sure if this necessitates any code changes in this repo. The only field that's actually defined in template.yaml is `id`, the rest were added when populating the tables.

**Review by this-andy — APPROVED** [review]

**Pull request closed without merging**